### PR TITLE
[Impeller] drawVertices uber shader.

### PR DESCRIPTION
**Pull request by jonahwilliams**
In order to land https://github.com/flutter/engine/pull/52303 , we need to finally fix the advanced blend draw vertices combo. Right now a ColorFilter is used for advanced blends which doesn't work if there are overlapping vertices.

See also: https://github.com/flutter/flutter/issues/145707

The issue was fixed for drawVertices/drawAtlas pipeline blends using the porterduff shader. This extends this to advanced blends, but since drawVertices/atlas with an advanced blend is uncommon and  because we don't 15 new shader variants, just add one special uber shader.

Part of https://github.com/flutter/flutter/issues/131345

### CODE REVIEW

File: impeller/entity/shaders/blending/vertices_uber.frag [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+out f16vec4 frag_color;
+
+// A shader that implements the required src/dst blending for drawVertices and
+// drawAtlas advanced blends without requiring an offscreen render pass. This is
+// done in a single shader to reduce the permutations of PSO needed at runtime
+// for rarely used features.
+void main() {
+  f16vec4 dst = IPHalfUnpremultiply(v_color);
+  f16vec4 src = IPHalfUnpremultiply(texture(texture_sampler, v_texture_coords));
+  f16vec3 blend_result =
```
Comment (jonahwilliams) [inline impeller/entity/shaders/blending/vertices_uber.frag:29]: If this is fast enough maybe we'll just do all the advanced blends this way 😏

Comment (chinmaygarde) [reply impeller/entity/shaders/blending/vertices_uber.frag:29]: Oh, I don't see the T880 in the Vulkan malioc results. Perhaps that is too old for Vulkan? Maybe we should find an older baseline than the Mali-G78.

**Review by chinmaygarde — APPROVED** [review]
Besides the observation about this bumping up against the limits what older Android devices are capable of, lgtm because that should be limited to GLES.

File: impeller/entity/shaders/blending/vertices_uber.frag (hunk 1 above)
Comment (jonahwilliams) [reply impeller/entity/shaders/blending/vertices_uber.frag:29]: We shrunk down the malioc to two models, a pixel and I think the moto g4 gpu

Comment (chinmaygarde) [reply impeller/entity/shaders/blending/vertices_uber.frag:29]: Hmm, if memory serves, the reasoning was that one would capture the high end on Vulkan and the other the low one with GLES. But that was when we were still focusing on GLES. Should we have two for Vulkan (and decide what the low one should be)?

In any case, don't want to distract you. This is good. There are no spills on our current Vulkan baseline.

Comment (jonahwilliams) [reply impeller/entity/shaders/blending/vertices_uber.frag:29]: That's a good idea , we should revisit the gpu list

**Pull request merged**